cloudevents/pydantic/strict_event.py

```diff
@@ -0,0 +1,46 @@
+#  Copyright 2018-Present The CloudEvents Authors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+import typing
+
+import pydantic
+
+from cloudevents.pydantic import event, event_attribute_types
+from cloudevents.pydantic.event import (
+    DATA_CONTENT_TYPE_FIELD,
+    DATA_SCHEMA_FIELD,
+    ID_FIELD,
+    SOURCE_FIELD,
+    SUBJECT_FIELD,
+    TYPE_FIELD,
+)
+
+
+class StrictCloudEvent(event.CloudEvent):
```

Comment:
As far as I see, this type is used only in the tests. What's the idea behind it?

Reply:
Ok, so that's where all that custom pyndatic types are used, right?

Reply:
IDK how I feel about this class. From one point I agree that it may be beneficial to have a "strict" version.

From the other, I'd better see all the required validations (like non-empty type and ID) in the main implementation.

Reply:
I was not sure about it myself. The main reason i chose to put them here is to not break the api presented by the http cloudevent. 
Im planning on creating a strict version for the http version as-well

Reply:
@xSAVIKx  what do you think about the consistency concerns? 
Also the "Strict" class concern has president in the redis python client library with the `StrictRedis` class

Reply:
I like the idea of having a contract overall, but am concerned about the consistency as well.

And I don't know if I like the `Strict` version. Especially taking into account that you were forced to externalize all the field declarations in order to avoid code duplication. I don't like this at all. This just makes code much less clear and harder to read.

Reply:
Very well, i will remove all this logic. 
If a user wish to create a strict version he may do it himself